tests/gnrc_tcp_client/main.c

```diff
@@ -39,12 +40,33 @@
 uint8_t bufs[CONNS][NBYTE];
 uint8_t stacks[CONNS][THREAD_STACKSIZE_DEFAULT + THREAD_EXTRA_STACKSIZE_PRINTF];
 
+/* "ifconfig" shell command */
+extern int _gnrc_netif_config(int argc, char **argv);
+
 void *cli_thread(void *arg);
+bool verify_ip_addr(gnrc_netif_t *netif);
 
 int main(void)
 {
+    gnrc_netif_t *netif;
+
+    if (!(netif = gnrc_netif_iter(NULL))) {
+        printf("No valid network interface found\n");
+        return -1;
+    }
+
+    /* Set pre-configured IP address */
+    char if_pid[] = {netif->pid + '0', '\0'};
+    char *cmd[] = {"ifconfig", if_pid, "add", "unicast", LOCAL_ADDR};
+    _gnrc_netif_config(5, cmd);
```

Comment:
I know you did this already in the server application, but why are you using the shell command for this? [Here's a nice function](http://doc.riot-os.org/group__net__gnrc__netif.html#gaf96707a5e322b5fa8458fba45de01837) that does exactly the same but much leaner without any string conversions.